packages/components/messages/src/warning-message/warning-message.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import Text from '@commercetools-uikit/text';
+
+type TIntlMessageDescriptor = {
+  id: string | number;
+  description?: string | object;
+  defaultMessage: string;
+};
+
+type TWarningMessageProps = {
+  intlMessage?: TIntlMessageDescriptor;
+  children?: React.ReactNode;
```

Comment:
A `React.ReactNode` can be `null` or `undefined`. As a result this can be `children: React.ReactNode` if I am not mistaken.

- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L237

Reply:
Yes, ReactNode can be `undefined`, but I recall getting type errors when not specifying `children?` because this is encapsulated into a custom specified type `TWarningMessageProps`

Reply:
Gotcha.

Reply:
Yes, the "optional" prop is not the same as the value being `undefined`.

<img width="585" alt="image" src="https://user-images.githubusercontent.com/1110551/107491809-f03ba900-6b8b-11eb-8a7a-e29f97dbfc23.png">